afs-storage-api/src/main/java/com/powsybl/afs/storage/AppStorageArchive.java

```diff
@@ -206,34 +213,36 @@ private void writeTimeSeries(NodeInfo nodeInfo, Path nodeDir) throws IOException
         Files.createDirectory(timeSeriesDir);
 
         for (TimeSeriesMetadata metadata : storage.getTimeSeriesMetadata(nodeInfo.getId(), timeSeriesNames)) {
-            String timeSeriesFileName = URLEncoder.encode(metadata.getName() + TSEXTENSION, StandardCharsets.UTF_8.name());
-            Path timeSeriesNameDir = timeSeriesDir.resolve(timeSeriesFileName);
-            Files.createDirectory(timeSeriesNameDir);
-
-            // write metadata
-            try (Writer writer = Files.newBufferedWriter(timeSeriesNameDir.resolve("metadata.json"), StandardCharsets.UTF_8)) {
-                objectWriter.writeValue(writer, metadata);
-            }
+            if (archiveDependencies.getkeepTSList().isEmpty() || archiveDependencies.getkeepTSList().get(nodeInfo.getPseudoClass())) {
```

Comment:
this not js. just check `archiveDependencies.getKeepTSList().contains(nodeInfo.getPseudoClass)`